arbnode/dataposter/data_poster.go

```diff
@@ -187,23 +193,36 @@ func (p *DataPoster[Meta]) GetNextNonceAndMeta(ctx context.Context) (uint64, Met
 
 const minRbfIncrease = arbmath.OneInBips * 11 / 10
 
-func (p *DataPoster[Meta]) getFeeAndTipCaps(ctx context.Context, gasLimit uint64, lastTipCap *big.Int, dataCreatedAt time.Time, backlogOfBatches uint64) (*big.Int, *big.Int, error) {
+func (p *DataPoster[Meta]) getFeeAndTipCaps(ctx context.Context, gasLimit uint64, lastFeeCap *big.Int, lastTipCap *big.Int, dataCreatedAt time.Time, backlogOfBatches uint64) (*big.Int, *big.Int, error) {
+	config := p.config()
 	latestHeader, err := p.headerReader.LastHeader(ctx)
 	if err != nil {
 		return nil, nil, err
 	}
+	newFeeCap := new(big.Int).Mul(latestHeader.BaseFee, big.NewInt(2))
+	newFeeCap = arbmath.BigMax(newFeeCap, arbmath.FloatToBig(config.MinFeeCapGwei*params.GWei))
+
 	newTipCap, err := p.client.SuggestGasTipCap(ctx)
 	if err != nil {
 		return nil, nil, err
 	}
+	newTipCap = arbmath.BigMax(newTipCap, arbmath.FloatToBig(config.MinTipCapGwei*params.GWei))
+
+	hugeTipIncrease := false
 	if lastTipCap != nil {
 		newTipCap = arbmath.BigMax(newTipCap, arbmath.BigMulByBips(lastTipCap, minRbfIncrease))
+		// hugeTipIncrease is true if the new tip cap is at least 10x the last tip cap
+		hugeTipIncrease = arbmath.BigDiv(newTipCap, lastTipCap).Cmp(big.NewInt(10)) >= 0
```

Comment:
it isn't possible for `lastTipCap` to be zero, is it?

Reply:
Hmm that's a good point, I'll add a safety check